aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -125,49 +197,51 @@ __global__ void cunn_SpatialSoftMaxForward(
       // I didn't want to thread an extra template parameter, and nvcc
       // seems to be smart enough to hoist the if outside of the loops.
       ////////////////////////////////////////////////////////////
+ 
       if (blockDim.x > 1) {
-        T max_input = THCNumerics<T>::min();
+        accscalar_t max_input = THCNumerics<accscalar_t>::min();
```

Comment:
Why do we keep max in accscalar now? That's not how it used to be

Reply:
1) comparison operators for half work on floats only anyway, so there will be conversions. Might as well convert in the beginning and keep intermediates in float
2) reduced max will still have to be converted to accscalar in epilogue, either when epilogue is constructed or for every call, so there's also nothing to be gained by keeping it in scalar